### ISSUE TITLE
[NCM] Web debugging support

### ISSUE THREAD

**Issue by itsjustkevin** [dart-lang/sdk#50735]

**Comment by itsjustkevin** [comment]
@sigmundch is this relevant for the feature?  If so, who would be the person to tackle this?

**Comment by a-siva** [comment]
On the VM side we implemented the ability to display this attribute for a class here
https://dart-review.googlesource.com/c/sdk/+/285360
The VM service API was modified for this and I presume the dwds would have to be changed to provide the same information.

**Comment by sigmundch** [comment]
Thanks for reaching out @itsjustkevin - @annagrin is the main contact here.

**Comment by itsjustkevin** [comment]
@annagrin for my understanding, are these features part of the sdk or in a package?

**Comment by sigmundch** [comment]
Most of it lives in the dwds package, which is outside the SDK. That said, now that devtools is linked through DEPS and released with the sdk, I'm not sure whether that adds some constraints. 

@annagrin - let us know to make sure we properly track whether the SDK hash can be selected prior to the support being completely landed or not. If this is not blocking, we could potentially move it to the next milestone, post the beta cutoff

**Comment by itsjustkevin** [comment]
Friendly ping.

**Comment by vsmenon** [comment]
@annagrin - anything to land here?

**Comment by vsmenon** [comment]
@annagrin @elliette - anything to do here or can we close?

**Comment by elliette** [comment]
Same as https://github.com/dart-lang/sdk/issues/51289, none of these new class attributes are currently being read by any of the debugging tools (DevTools, dap). Therefore this can be follow-up for after Dart 3.0.0.

**Comment by vsmenon** [comment]
Thanks!

@itsjustkevin - what do you think about moving these two to a 3.1 milestone?

**Comment by itsjustkevin** [comment]
moving this to the cleanup milestone to be done after the 3.0 release.

**Comment by eernstg** [comment]
@elliette, based on 'Same as #51289' which is closed, we might be able to close this issue?

**Comment by a-siva** [comment]
> @elliette, based on 'Same as #51289' which is closed, we might be able to close this issue?

I believe the comment in #51289 states there is nothing to do at the moment but something to consider for the future if the developers tool creates the UI to display this information. considering that would make sense to treat this as the issue for tracking that feature ?

**Comment by elliette** [comment]
> I believe the comment in #51289 states there is nothing to do at the moment but something to consider for the future if the developers tool creates the UI to display this information. considering that would make sense to treat this as the issue for tracking that feature ?

Yes, keeping this open to track implementation on web sounds good. We haven't prioritized implementing this in DWDS because currently we aren't exposing class modifier information in any of the debugging tools.
